conans/model/requires.py

```diff
@@ -317,6 +317,10 @@ def deduce_package_id_mode(self, pkg_type, dep_pkg_type, lib_mode, bin_mode, bui
                 self.package_id_mode = build_mode
             return  # At the moment no defaults
 
+        if pkg_type is PackageType.HEADER:
```

Comment:
Do we know what is `dep_pkg_type` here? is it UNKNOWN? Could we assert it?

Reply:
It doesn't matter. As long as you are header-only, your transitive dependencies do not affect, they can be header, static-lib, shared-lib... they will still not affect the header-only artifacts.